docs/proposals/compatibility-mode.md

```diff
@@ -0,0 +1,92 @@
+
+## Goals
+
+- Provide different options to allow users to customize the manifest uploading behavior to the registry
+- Provide an easy-to-use and secure user experience when switching the behaviors
+- Enable ORAS to work with more registries flexibly 
+
+## Solution
+
+Adding new flags to `oras push` and `oras attach` respectively with different variables to configure the manifest uploading behaviors. 
+
+- Adding a flag `--image-spec` to `oras push` and `oras attach` to force uploading a specific manifest type to registry
+- Adding a flag `--distribution-spec` to `oras attach`, `oras attach`, `oras cp`, and `oras manifest push` to configure compatibility with registry when pushing or copying an image/artifact manifest. This flag is also applicable to `oras discover` for filtering the referrers.
+
+### Force uploading a specific manifest type using a flag `--image-spec`
+
+It follows `--image-spec <spec version>-<manifest type>` to enable configuration of using which spec version and manifest type. Currently, it only supports specifying v1.1 as the spec version. 
+
+| registry support                        | v1.1-artifact | v1.1-image | 
+| :-------------------------------------- | ----------------- | -------------- | 
+| OCI spec 1.0                            | no                | yes |
+| OCI spec 1.1 without referrers API      | yes               | yes            | 
+| OCI spec 1.1 with referrers API support | yes               | yes            | 
+
+If users want to force pushing a specific version of OCI artifact manifest to a registry, they can use `--image-spec v1.1-artifact`. An OCI artifact manifest will be packed and uploaded. Users might choose it for security requirements, such as pushing a signature to a registry without changing its digest. For example:
+
+```bash
+oras push localhost:5000/hello-artifact:v1 \
+--image-spec v1.1-artifact
+--artifact-type sbom/example \
+sbom.json 
+```
+
+If users want to force pushing an OCI image manifest, no matter whether the registry is compliant with the OCI Spec v1.0 or v1.1, using `--image-spec v1.1-image` will only upload the OCI image manifest to a registry. This option is useful when users have concerns to use OCI artifact manifest or need to migrate content to OCI v1.0 compliant registry. For example:
+
+```bash
+oras push localhost:5000/hello-artifact:v1 \
+--image-spec v1.1-image \
+--artifact-type sbom/example \
+sbom.json \
+```
+
+### Configure compatibility with OCI registry using a flag `--distribution-spec`
+
+Based on the Referrers API status in the registry, users can use flag `--distribution-spec <spec version>-<api>-<option>` to configure compatibility with registry. 
+
+| registry support                        |  v1.1-referrers-api | v1.1-referrers-tag |
+| :-------------------------------------- | --- | --- | 
+| OCI spec 1.0                            | no  | yes[^footnote] |
+| OCI spec 1.1 without referrers API      | no  | yes |
```

Comment:
`OCI spec 1.0` should be exactly the same as `OCI spec 1.1 without referrers API` since we don't do any detection.

Reply:
Of course, if nothing specified. It will go into an `auto` mode, and then it requires `[^footnote]`.